client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -421,4 +421,13 @@ void checkpoint(String checkpointName, PositionInternal lastPosition) throws Rei
             throw new ReinitializationRequiredException();
         }
     }
+
+    public String refreshDelegationToken(Segment segmentId) {
+        synchronized (this) {
+            if (this.latestDelegationToken == null ) {
+                this.latestDelegationToken = getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segmentId.getScope(), segmentId.getStreamName()), RuntimeException::new);
```

Comment:
Since a ReaderGroup can read from multiple streams and a reader can be assigned segments for any streams managed by the reader group,  do we need to have multiple delegation tokens ?

Reply:
Dropped caching the `delegationToken` in the `groupstate` as now more than one stream shares the state and `delegationToken` is stream level.

Reply:
As enhancement we could  cache the `delegationToken` in `groupState` since a `delegationToken` is on per Stream level basis. 

The advantages of caching include:
- No additional controller api call to fetch token for every segmentTruncationExcecption recieved by Reader on its managed segments.
- No additional controller api call to fetch token for every segment acquired by the reader, when a new instance of `SegmentInputStream` is created inside `EventStreamReaderImpl.acquireSegmentsIfNeeded()`.
- Reuse the `delegationToken` returned by `controller.getSuccessors()` api.